src/main/scala/com/evolutiongaming/retry/Sleep.scala

```diff
@@ -0,0 +1,24 @@
+package com.evolutiongaming.retry
+
+import cats.effect.Clock
+import cats.effect.GenTemporal
+import scala.concurrent.duration.FiniteDuration
+
+trait Sleep[F[_]] extends Clock[F] {
+
+  def sleep(time: FiniteDuration): F[Unit]
+
+}
+object Sleep {
+
+  def apply[F[_]](implicit F: Sleep[F]): Sleep[F] = F
+
+  implicit def fromGenTemporal[F[_]](implicit F: GenTemporal[F, _]): Sleep[F] =
+    new Sleep[F] {
+      def applicative = F.applicative
```

Comment:
yo don't need those methods, right?

Reply:
Yes, only `sleep` and `instant` is used.